Lib/test/test_embed.py

```diff
@@ -1293,24 +1294,26 @@ def test_init_pybuilddir(self):
             # directory (tmpdir)
             subdir = 'libdir'
             libdir = os.path.join(tmpdir, subdir)
+            # The stdlib dir is dirname(executable) + VPATH + 'Lib'
+            stdlibdir = os.path.join(tmpdir, 'Lib')
             os.mkdir(libdir)
 
             filename = os.path.join(tmpdir, 'pybuilddir.txt')
             with open(filename, "w", encoding="utf8") as fp:
                 fp.write(subdir)
 
             module_search_paths = self.module_search_paths()
+            module_search_paths[-2] = stdlibdir
             module_search_paths[-1] = libdir
 
             executable = self.test_exe
             config = {
+                'base_exec_prefix': tmpdir,
+                'base_prefix': tmpdir,
                 'base_executable': executable,
                 'executable': executable,
                 'module_search_paths': module_search_paths,
-                # The current getpath.c doesn't determine the stdlib dir
-                # in this case.
-                'stdlib_dir': None,
-                'use_frozen_modules': -1,
```

Comment:
This implies that the behavior changed.  I'm not opposed to improving it, but I thought this PR was just about rewriting getpath.c in pure Python.

Reply:
Same behaviour change as elsewhere: we now more reliably know whether we're in a build directory or not.